File: LeetCode/75.c
void sortColors(int* nums, int numsSize) {
    int w = 0, r = 0, b = numsSize -1;

    while ( w <= b)
    {
        if (nums[w] == 0)
        {
            int temp = nums[w];
            nums[w] = nums[r];
            nums[r] = temp;
            w++; r++;
        }
        else if (nums[w] == 1)
            w++;

        else
        {
            int temp = nums[w];
            nums[w] = nums[b];
            nums[b] = temp;
            b--;
        }
    }
}

File: LeetCode/75.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        r, w , b = 0,0, len(nums) -1

        while w <= b:
            if nums[w] == 0:
                nums[w], nums[r] = nums[r], nums[w]
                w += 1
                r += 1
            
            elif nums[w] == 1:
                w+= 1

            else :
                nums[w] , nums[b] = nums[b], nums[w]
                b -= 1
            
        
